domain/status/service/storagestatus_test.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/juju/tc"
+
+	corestatus "github.com/juju/juju/core/status"
+	"github.com/juju/juju/domain/status"
+)
+
+type storageStatusSuite struct {
+	now time.Time
+}
+
+func TestStorageStatusSuite(t *testing.T) {
+	tc.Run(t, &storageStatusSuite{})
+}
+
+func (s *storageStatusSuite) SetUpTest(c *tc.C) {
+	s.now = time.Now()
+}
+
+func (s *storageStatusSuite) TestEncodeFilesystemStatus(c *tc.C) {
+	testCases := []struct {
+		input  corestatus.StatusInfo
+		output status.StatusInfo[status.StorageFilesystemStatusType]
+	}{
+		{
+			input: corestatus.StatusInfo{
+				Status: corestatus.Pending,
+			},
+			output: status.StatusInfo[status.StorageFilesystemStatusType]{
+				Status: status.StorageFilesystemStatusTypePending,
+			},
+		},
+		{
+			input: corestatus.StatusInfo{
+				Status: corestatus.Error,
+			},
+			output: status.StatusInfo[status.StorageFilesystemStatusType]{
+				Status: status.StorageFilesystemStatusTypeError,
+			},
+		},
+		{
+			input: corestatus.StatusInfo{
+				Status: corestatus.Attaching,
+			},
+			output: status.StatusInfo[status.StorageFilesystemStatusType]{
+				Status: status.StorageFilesystemStatusTypeAttaching,
+			},
+		},
+		{
+			input: corestatus.StatusInfo{
+				Status: corestatus.Attached,
+			},
+			output: status.StatusInfo[status.StorageFilesystemStatusType]{
+				Status: status.StorageFilesystemStatusTypeAttached,
+			},
+		},
+		{
+			input: corestatus.StatusInfo{
+				Status: corestatus.Detaching,
+			},
+			output: status.StatusInfo[status.StorageFilesystemStatusType]{
+				Status: status.StorageFilesystemStatusTypeDetaching,
+			},
+		},
+		{
+			input: corestatus.StatusInfo{
+				Status: corestatus.Detached,
+			},
+			output: status.StatusInfo[status.StorageFilesystemStatusType]{
+				Status: status.StorageFilesystemStatusTypeDetached,
+			},
+		},
+		{
+			input: corestatus.StatusInfo{
+				Status: corestatus.Destroying,
+			},
+			output: status.StatusInfo[status.StorageFilesystemStatusType]{
+				Status: status.StorageFilesystemStatusTypeDestroying,
+			},
+		},
+	}
+
+	for i, test := range testCases {
+		c.Run(fmt.Sprintf("Test %d", i), func(t *testing.T) {
```

Comment:
This should be the status under test from the input. The i value isn't actually  useful. So this would just become. `c.Run(test.input.State, func(t *testing.T)`